homeassistant/components/totalconnect/__init__.py

```diff
@@ -24,30 +26,56 @@
     extra=vol.ALLOW_EXTRA,
 )
 
-TOTALCONNECT_PLATFORMS = ["alarm_control_panel", "binary_sensor"]
+PLATFORMS = ["alarm_control_panel", "binary_sensor"]
+
 
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Old way to set up."""
+    return True
 
-def setup(hass, config):
-    """Set up TotalConnect component."""
-    conf = config[DOMAIN]
 
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up upon config entry in user interface."""
+    hass.data.setdefault(DOMAIN, {})
+
+    conf = entry.data
     username = conf[CONF_USERNAME]
     password = conf[CONF_PASSWORD]
 
-    client = TotalConnectClient.TotalConnectClient(username, password)
+    client = await hass.async_add_executor_job(
+        TotalConnectClient.TotalConnectClient, username, password
+    )
 
-    if client.token is False:
+    if not client.is_logged_in():
         _LOGGER.error("TotalConnect authentication failed")
         return False
 
-    hass.data[DOMAIN] = TotalConnectSystem(username, password, client)
+    hass.data[DOMAIN][username] = TotalConnectSystem(username, password, client)
```

Comment:
Instead of username, let's use `entry.entry_id`
```suggestion
    hass.data[DOMAIN][entry.entry_id] = TotalConnectSystem(username, password, client)
```